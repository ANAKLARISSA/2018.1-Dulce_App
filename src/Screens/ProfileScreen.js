import React from 'react';
import {View , Text} from 'react-native';
import axios from 'axios';
import store from '../Reducers/store';
import {Container, Content, Spinner} from 'native-base';
import SmallLogo from '../Components/SmallLogo';
import ScaleIcon from '../Components/ScaleIcon';

const styles = {
  container: {
    flex: 1,
    flexDirection: 'column',
    padding: 20,
    backgroundColor: '#FFF'

  },
  text: {
    fontSize: 20,
    marginBottom: 15,
    alignSelf: 'center'
  },
  name: {
    fontSize: 30,
    marginBottom: 15,
    alignSelf: 'center'
  },
  informacoes: {
    alignSelf: 'center'
  },
  icon: {
    justifyContent: 'center',
    alignItems: 'center'
  }
};

class ProfileScreen extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      profile: {},
      loading: true
    };
  }

  componentDidMount() {
    this.setState({loading: true});
    const url = 'http://172.18.0.1:8083/api/userManager/listById/?id=' + this.props.navigation.state.params.userId;
    axios.get(url,{
      headers: {
        'x-access-token': store.getState().currentUser.token
      }
    })
      .then((response) => {this.setState({profile: response.data,loading: false});});
  }

  render() {
    return (
      <View style={styles.container}>
        {
          this.state.loading ? (
            <Container>
              <Content>
                <Spinner color='purple'/>
              </Content>
            </Container>
          ) : (
            <View style={styles.informacoes}>
              <SmallLogo />
              <Text style={styles.name}>{this.state.profile.name}</Text>
              <Text style={styles.text}>Matr√≠cula:{this.state.profile.registration}</Text>
              <Text style={styles.text}>Setor: {this.state.profile.sector}</Text>
              <Text style={styles.text}>Hospital: {this.state.profile.hospital}</Text>
              <View style = {styles.icon}>
                <ScaleIcon onPress = {() => null} />
              </View>
            </View>
          )
        }
      </View>
    );
  }
}

export default ProfileScreen;
