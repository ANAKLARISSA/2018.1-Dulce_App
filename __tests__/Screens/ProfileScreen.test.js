import React from 'react';
import renderer from 'react-test-renderer';
import ProfileScreen from '../../src/Screens/ProfileScreen';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import {shallow} from 'enzyme';
require('bezier');
Enzyme.configure({adapter: new Adapter()});

const props = {

  navigation: {
    state: {
      params: {
        userId: 1
      }
    }
  }
};
test('renders correctly' , () => {
  const tree = renderer
  .create(<ProfileScreen {...props} />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});

const flushPromises = () => new Promise(resolve => setImmediate(resolve));

it('testing axios', async () => {
  const wrapper = shallow(<ProfileScreen {...props}/>);
  await flushPromises();
  wrapper.update();
  expect(wrapper).toMatchSnapshot();
});

// it('testing onPress null', async() => {
//   const wrapper = shallow(<ProfileScreen {...props}/>);
//   await flushPromises();
//   wrapper.update();
//   // const scale = wrapper.find('ScaleItem').at(0);
//   expect(undefined).toBe(undefined);
// });
