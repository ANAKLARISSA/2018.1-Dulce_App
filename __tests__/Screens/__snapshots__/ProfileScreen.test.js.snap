// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
      "flex": 1,
      "flexDirection": "column",
      "padding": 20,
    }
  }
>
  <View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F8F8F8",
            "borderBottomColor": "#a7a6ab",
            "borderBottomWidth": 0.5,
            "elevation": 3,
            "flexDirection": "row",
            "height": 64,
            "justifyContent": "center",
            "left": 0,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 15,
            "right": 0,
            "shadowColor": undefined,
            "shadowOffset": undefined,
            "shadowOpacity": undefined,
            "shadowRadius": undefined,
            "top": 0,
          },
          Object {
            "backgroundColor": "#5f4b8b",
          },
        ]
      }
      virtual={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 17,
                "fontWeight": "600",
                "textAlign": "center",
              },
              Object {
                "fontFamily": "Raleway",
              },
            ]
          }
          virtual={undefined}
        >
           
          Meu perfil
           
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "height": 1334,
      }
    }
    virtual={undefined}
  >
    <RCTScrollView
      automaticallyAdjustContentInsets={false}
      contentContainerStyle={
        Array [
          Object {
            "padding": undefined,
          },
          undefined,
        ]
      }
      contentInset={
        Object {
          "bottom": 0,
        }
      }
      enableAutomaticScroll={true}
      enableResetScrollToCoords={true}
      extraHeight={75}
      extraScrollHeight={0}
      getScrollResponder={[Function]}
      handleOnScroll={[Function]}
      keyboardDismissMode="interactive"
      keyboardOpeningTime={250}
      keyboardShouldPersistTaps="handled"
      keyboardSpace={0}
      onScroll={[Function]}
      resetKeyboardSpace={[Function]}
      resetScrollToCoords={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      scrollEventThrottle={1}
      scrollForExtraHeightOnAndroid={[Function]}
      scrollToEnd={[Function]}
      scrollToFocusedInput={[Function]}
      scrollToPosition={[Function]}
      showsVerticalScrollIndicator={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        }
      }
      viewIsInsideTabBar={false}
      virtual={undefined}
    >
      <View>
        <ActivityIndicator
          animating={true}
          color="#5f4b8b"
          hidesWhenStopped={true}
          size="large"
          style={
            Object {
              "height": 80,
            }
          }
          virtual={undefined}
        />
      </View>
    </RCTScrollView>
     
  </View>
</View>
`;

exports[`testing axios 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProfileScreen
    navigation={
      Object {
        "state": Object {
          "params": Object {
            "userId": 1,
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ScreenHeader
          title="Meu perfil"
        />,
        <View
          style={
            Object {
              "alignSelf": "center",
            }
          }
        >
          <SmallLogo />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "fontSize": 30,
                "marginBottom": 15,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "fontSize": 20,
                "marginBottom": 15,
              }
            }
          >
            Matrícula:
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "fontSize": 20,
                "marginBottom": 15,
              }
            }
          >
            Setor: 
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "fontSize": 20,
                "marginBottom": 15,
              }
            }
          >
            Hospital: 
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <ScaleIcon
              onPress={[Function]}
            />
          </View>
        </View>,
      ],
      "style": Object {
        "backgroundColor": "#FFF",
        "flex": 1,
        "flexDirection": "column",
        "padding": 20,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "title": "Meu perfil",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <SmallLogo />,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 30,
                  "marginBottom": 15,
                }
              }
            />,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 20,
                  "marginBottom": 15,
                }
              }
            >
              Matrícula:
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 20,
                  "marginBottom": 15,
                }
              }
            >
              Setor: 
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 20,
                  "marginBottom": 15,
                }
              }
            >
              Hospital: 
            </Text>,
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                }
              }
            >
              <ScaleIcon
                onPress={[Function]}
              />
            </View>,
          ],
          "style": Object {
            "alignSelf": "center",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": undefined,
              "ellipsizeMode": "tail",
              "style": Object {
                "alignSelf": "center",
                "fontSize": 30,
                "marginBottom": 15,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                "Matrícula:",
                undefined,
              ],
              "ellipsizeMode": "tail",
              "style": Object {
                "alignSelf": "center",
                "fontSize": 20,
                "marginBottom": 15,
              },
            },
            "ref": null,
            "rendered": Array [
              "Matrícula:",
              undefined,
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                "Setor: ",
                undefined,
              ],
              "ellipsizeMode": "tail",
              "style": Object {
                "alignSelf": "center",
                "fontSize": 20,
                "marginBottom": 15,
              },
            },
            "ref": null,
            "rendered": Array [
              "Setor: ",
              undefined,
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                "Hospital: ",
                undefined,
              ],
              "ellipsizeMode": "tail",
              "style": Object {
                "alignSelf": "center",
                "fontSize": 20,
                "marginBottom": 15,
              },
            },
            "ref": null,
            "rendered": Array [
              "Hospital: ",
              undefined,
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <ScaleIcon
                onPress={[Function]}
              />,
              "style": Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onPress": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ScreenHeader
            title="Meu perfil"
          />,
          <View
            style={
              Object {
                "alignSelf": "center",
              }
            }
          >
            <SmallLogo />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 30,
                  "marginBottom": 15,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 20,
                  "marginBottom": 15,
                }
              }
            >
              Matrícula:
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 20,
                  "marginBottom": 15,
                }
              }
            >
              Setor: 
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 20,
                  "marginBottom": 15,
                }
              }
            >
              Hospital: 
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                }
              }
            >
              <ScaleIcon
                onPress={[Function]}
              />
            </View>
          </View>,
        ],
        "style": Object {
          "backgroundColor": "#FFF",
          "flex": 1,
          "flexDirection": "column",
          "padding": 20,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "title": "Meu perfil",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <SmallLogo />,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "center",
                    "fontSize": 30,
                    "marginBottom": 15,
                  }
                }
              />,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "center",
                    "fontSize": 20,
                    "marginBottom": 15,
                  }
                }
              >
                Matrícula:
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "center",
                    "fontSize": 20,
                    "marginBottom": 15,
                  }
                }
              >
                Setor: 
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "center",
                    "fontSize": 20,
                    "marginBottom": 15,
                  }
                }
              >
                Hospital: 
              </Text>,
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                  }
                }
              >
                <ScaleIcon
                  onPress={[Function]}
                />
              </View>,
            ],
            "style": Object {
              "alignSelf": "center",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": undefined,
                "ellipsizeMode": "tail",
                "style": Object {
                  "alignSelf": "center",
                  "fontSize": 30,
                  "marginBottom": 15,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": Array [
                  "Matrícula:",
                  undefined,
                ],
                "ellipsizeMode": "tail",
                "style": Object {
                  "alignSelf": "center",
                  "fontSize": 20,
                  "marginBottom": 15,
                },
              },
              "ref": null,
              "rendered": Array [
                "Matrícula:",
                undefined,
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": Array [
                  "Setor: ",
                  undefined,
                ],
                "ellipsizeMode": "tail",
                "style": Object {
                  "alignSelf": "center",
                  "fontSize": 20,
                  "marginBottom": 15,
                },
              },
              "ref": null,
              "rendered": Array [
                "Setor: ",
                undefined,
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": Array [
                  "Hospital: ",
                  undefined,
                ],
                "ellipsizeMode": "tail",
                "style": Object {
                  "alignSelf": "center",
                  "fontSize": 20,
                  "marginBottom": 15,
                },
              },
              "ref": null,
              "rendered": Array [
                "Hospital: ",
                undefined,
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <ScaleIcon
                  onPress={[Function]}
                />,
                "style": Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "onPress": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
